### PR TITLE
Avoid checking the class loading constraints with -Xverify:none

### PR DESCRIPTION
**Pull request by ChengJin01**
The change is to avoid checking the class loading constraints
when the verifier is disabled by -Xverify:none.

Close: #4904

Signed-off-by: Cheng Jin <jincheng@ca.ibm.com>

### CODE REVIEW

**Comment by ChengJin01** [comment]
The fix was verified by the external user.

Reviewer: @DanHeidinga 
FYI: @pshipton

File: runtime/vm/createramclass.cpp [hunk 1]
```diff
@@ -1250,7 +1253,7 @@ processVTableMethod(J9VMThread *vmThread, J9ClassLoader *classLoader, UDATA *vTa
 						}
 						/* fill in vtable, override parent slot */
 						J9ClassLoader *superclassVTableMethodLoader = superclassVTableMethodClass->classLoader;
-						if (superclassVTableMethodLoader != classLoader) {
+						if ((superclassVTableMethodLoader != classLoader) && verifierLoaded) {
```
Comment (DanHeidinga) [inline runtime/vm/createramclass.cpp]: Please move the check out to include line 1255 as well.   I'd rather deal with an extra level indenting if it gives simpler to reason about code

File: runtime/vm/createramclass.cpp [hunk 2]
```diff
@@ -601,7 +603,7 @@ addInterfaceMethods(J9VMThread *vmStruct, J9ClassLoader *classLoader, J9Class *i
 							if (NULL != methodClass) {
 								vTableMethodLoader = methodClass->classLoader;
 							}
-							if (interfaceLoader != vTableMethodLoader) {
+							if ((interfaceLoader != vTableMethodLoader) && verifierLoaded) {
```
[the remark this answers is not in the document]
Comment (ChengJin01) [reply runtime/vm/createramclass.cpp]: Agreed and moved it to line 601 as suggested.

File: runtime/vm/createramclass.cpp [hunk 3]
```diff
@@ -454,6 +454,8 @@ addInterfaceMethods(J9VMThread *vmStruct, J9ClassLoader *classLoader, J9Class *i
 	J9Method **vTableMethods = J9VTABLE_FROM_HEADER(vTableAddress);
 	J9ROMClass *interfaceROMClass = interfaceClass->romClass;
 	UDATA count = interfaceROMClass->romMethodCount;
+	bool verifierLoaded = J9_ARE_ANY_BITS_SET(vmStruct->javaVM->runtimeFlags, J9_RUNTIME_VERIFY);
```
[the remark this answers is not in the document]
Comment (ChengJin01) [reply runtime/vm/createramclass.cpp]: Already renamed to `verifierEnabled`

File: runtime/vm/createramclass.cpp (hunk 1 above)
Comment (ChengJin01) [reply runtime/vm/createramclass.cpp]: Already adjusted as suggested.

**Comment by DanHeidinga** [comment]
Jenkins test sanity zlinux jdk8

**Comment by DanHeidinga** [comment]
@ChengJin01 Can you create a PR for this change against the v0.14.0-release branch as well as we've already split for that release.

**Comment by ChengJin01** [comment]
@DanHeidinga ,  the PR against 0.14.0 has been created at #5272.

**Pull request merged**